core/services/aws/bloodDonation/acceptDonationRequest.ts

```diff
@@ -45,88 +45,189 @@ async function acceptDonationRequestLambda(
   event: AcceptDonationRequestAttributes
 ): Promise<APIGatewayProxyResult> {
   try {
-    const userProfile = await userService.getUser(
-      event.donorId,
-      new DynamoDbTableOperations<UserDetailsDTO, UserFields, UserModel>(new UserModel())
+    const { donorId, seekerId, requestPostId, createdAt, status } = event
+
+    const acceptanceRecord = await acceptDonationService.getAcceptanceRecord(
+      seekerId,
+      requestPostId,
+      donorId,
+      new AcceptedDonationDynamoDbOperations<
+      AcceptedDonationDTO,
+      AcceptedDonationFields,
+      AcceptDonationRequestModel
+      >(new AcceptDonationRequestModel())
     )
-    if (userProfile === null) {
-      throw new Error('Cannot find user')
+    if (acceptanceRecord !== null && acceptanceRecord.status === AcceptDonationStatus.COMPLETED) {
+      throw new Error('You already donated.')
     }
-    const donationPost = await bloodDonationService.getDonationRequest(
-      event.seekerId,
-      event.requestPostId,
-      event.createdAt,
-      new BloodDonationDynamoDbOperations<DonationDTO, DonationFields, BloodDonationModel>(
-        new BloodDonationModel()
+
+    if (status === AcceptDonationStatus.ACCEPTED && acceptanceRecord === null) {
+      await acceptBloodDonationRequest(donorId, seekerId, requestPostId, createdAt, status)
+    } else if (status === AcceptDonationStatus.IGNORED && acceptanceRecord !== null) {
```

Comment:
else if without an else?

Reply:
updated in f55b70d04d0d650f746b701e4740aa308ba5ee77